server/src/main/java/org/candlepin/sync/Importer.java

```diff
@@ -741,4 +870,33 @@ public void importContentDeliveryNetworks(File[] cdnFiles) throws IOException {
         }
         importer.store(cdns);
     }
+
+    private ImportUpstreamConsumer createImportUpstreamConsumer(Owner owner, UpstreamConsumer uc) {
+        ImportUpstreamConsumer iup = null;
+        if (uc == null && owner != null) {
+            uc = owner.getUpstreamConsumer();
+        }
+
+        // The owner may not have the upstream consumer set so we need
+        // to check.
+        if (uc != null) {
+            iup = new ImportUpstreamConsumer(uc);
+        }
+        return iup;
+    }
+
+    private File unpackExportFile(String fileName, InputStream exportInputStream)
+        throws ImportExtractionException {
+        try {
+            File tmpDir = new SyncUtils(config).makeTempDir("import");
+            extractArchive(tmpDir, fileName, exportInputStream);
+            return tmpDir;
+        }
+        catch (IOException e) {
+            log.error("Unable to extract export archive", e);
+            throw new ImportExtractionException(
+                i18n.tr("Unable to extract export archive"), e);
+        }
+    }
+
 }
```

Comment:
DId I miss the extension data extraction? If we don't process it out, doesn't it get erased with the manifest?